app-office/visidata/visidata-2.1.1.ebuild

```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7..9} )
+
+inherit distutils-r1 eutils
+
+DESCRIPTION="Terminal spreadsheet multitool for discovering and arranging data"
+HOMEPAGE="http://visidata.org"
+SRC_URI="https://github.com/saulpw/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="dev-python/python-dateutil[${PYTHON_USEDEP}]"
+DEPEND=""
+BDEPEND="
+	test? (
+		dev-vcs/git
+		dev-python/h5py[${PYTHON_USEDEP}]
+		dev-python/openpyxl[${PYTHON_USEDEP}]
+		dev-python/pandas[${PYTHON_USEDEP}]
+		dev-python/requests[${PYTHON_USEDEP}]
+		$(python_gen_impl_dep sqlite)
+		${RDEPEND}
```

Comment:
According to https://github.com/saulpw/visidata/blob/stable/dev/requirements-dev.txt this should also depend on pytest. The `python_enable_tests pytest` function will automatically add this dependency, and will also automatically add `IUSE="test"` and `RESTRICT="!test? ( test )"` so you can than remove those lines as well.

Reply:
`RDEPENDS` is also not needed, it is automatically assumed when doing tests.

Reply:
> this should also depend on pytest

Ah, I missed this, as I only `diff`ed the `requirements.txt` from the top-level directory with the version from the previous ebuild. If I had `diff`ed this sub-`requirements.txt` I would have seen the new dep. I'll make sure to add it.

> `RDEPENDS` is also not needed,

Hm, this is also an artifact of the original ebuild. I have no problem taking it out.

Reply:
> `RDEPENDS` is also not needed, it is automatically assumed when doing tests.

hm, actually, is this true? I see that [RDEPEND is added automatically](https://dev.gentoo.org/~mgorny/python-guide/distutils.html#using-distutils-enable-tests) when `distutils_enable_tests pytest` is used, but that's not being used here.

Reply:
you should use it and then remove RDEPEND.

Reply:
but how can I use `distutils_enable_tests` if I need to do all this weird `git init` stuff and then manually execute `dev/tests.sh` to run tests?

Reply:
`distutils_enable_tests` provides a default impl of `python_test` (in addition to doing the other things of IUSE and DEPEND/RDEPEND). As you are overriding the function, distutils_enable_tests wont change the python_test for you.
But you can still take advantage of the other things done by `distutils_enable_tests`, which are the dependency additions and USE flag fixing.

Reply:
So this is the preferred approach? It seems a bit roundabout, *especially* since `distutils_enable_tests` requires a target - what should I do, pick a random one? `distutils_enable_tests pytest`? But this is misleading to future maintainers - I'm not using `pytest`, I'm using a custom test script.

Reply:
>  I'm not using `pytest`, I'm using a custom test script.

I'm pretty sure the custom test script calls pytest at some point, since pytest is specified as a dependency by upstream after all.

Reply:
> > I'm not using `pytest`, I'm using a custom test script.
> 
> I'm pretty sure the custom test script calls pytest at some point, since pytest is specified as a dependency by upstream after all.

Indeed it does:
```
andrew@andrew-gentoo-pc visidata-stable % grep -r pytest
visidata/tests/test_path.py:import pytest
visidata/tests/test_edittext.py:import pytest
visidata/tests/test_edittext.py:    @pytest.fixture(autouse=True, scope='function')
visidata/tests/test_edittext.py:    @pytest.mark.parametrize('keys, result, kwargs', [
visidata/tests/test_edittext.py:            with pytest.raises(exception):
visidata/tests/test_commands.py:import pytest
visidata/tests/test_commands.py:@pytest.mark.usefixtures('curses_setup')
visidata/tests/conftest.py:import pytest
visidata/tests/conftest.py:@pytest.fixture(scope="class")
visidata/tests/conftest.py:@pytest.fixture(scope="function")
```